File: data/translated_projects/commons-fileupload/src/main/org/apache/commons/fileupload/FileCountLimitExceededException.py
from __future__ import annotations
import io
from src.main.org.apache.commons.fileupload.FileUploadException import *
import java # type: ignore
from src.main.org.apache.commons.fileupload.java_handler import *
class FileCountLimitExceededException(FileUploadException, metaclass=StaticFieldRedirector):
    def getLimit(self) -> int:

        idMapPyToJ = self.javaObj.pyToJ()
        
        idMapJToPy = dict()
        try:
            val = JavaHandler.mapping(self.javaObj.getLimit(), id_map=idMapJToPy)
            return val
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            self.javaObj.jToPy(idMapJToPy)
            
    def __init__(self, message: str, limit: int) -> None:

        idMapPyToJ = JavaHandler.valueToObject(dict(), "Map")
        translatedArg0 = JavaHandler.valueToObject(message, "String", idMapPyToJ)
        translatedArg1 = JavaHandler.valueToObject(limit, "long", idMapPyToJ)
        idMapJToPy = dict()
        try:
            self.javaObj = FileCountLimitExceededException.javaClz(translatedArg0, translatedArg1)
            self.javaObj.setPythonObject(self)
            self.javaObj.jToPy(idMapJToPy)
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            JavaHandler.mapping(translatedArg0, idMapJToPy, message)
            JavaHandler.mapping(translatedArg1, idMapJToPy, limit)
    javaClz = java.type("org.apache.commons.fileupload.FileCountLimitExceededException")
    def __getattr__(self, name):
        try:
            return object.__getattribute__(self, name)
        except AttributeError:
            pass
        return JavaHandler.mapping(getattr(self.javaClz, StaticFieldRedirector.unmangle_name(name)))

File: logs/glue/commons-fileupload/src/main/java/org/apache/commons/fileupload/FileCountLimitExceededException.java
package org.apache.commons.fileupload;

import org.apache.commons.fileupload.ContextInitializer;
import org.apache.commons.fileupload.ExceptionHandler;
import org.apache.commons.fileupload.IntegrationUtils;
import org.graalvm.polyglot.PolyglotException;
import org.graalvm.polyglot.Value;

public class FileCountLimitExceededException extends FileUploadException {
  private transient Value obj =
      IntegrationUtils.createDefaultPythonObject(
          ContextInitializer.getPythonClass(
              "main/org/apache/commons/fileupload/FileCountLimitExceededException.py",
              "FileCountLimitExceededException"));
  public static final long serialVersionUID = 6904179610227521789L;
  public final long limit;

  public Value getPythonObject() {
    return obj;
  }

  public void setPythonObject(Value obj) {
    this.obj = obj;
  }

  public long getLimit() {

    return limit;
  }

  public FileCountLimitExceededException(final String message, final long limit) {

    super(message, null);
    this.limit = limit;
  }

  public java.util.Map pyToJ() {
    java.util.Map idMap = new java.util.HashMap();
    return pyToJ(idMap);
  }

  public java.util.Map pyToJ(java.util.Map idMap) {
    super.pyToJ(idMap);

    return idMap;
  }

  public Value jToPy() {
    Value idMap = IntegrationUtils.mapToPython(new java.util.HashMap());
    return jToPy(idMap);
  }

  public Value jToPy(Value idMap) {
    super.jToPy(idMap);
    this.obj.invokeMember("__setattr__", "javaObj", this);
    this.obj.invokeMember(
        "__setattr__",
        "_FileCountLimitExceededException__limit",
        IntegrationUtils.mapToPython(
            limit, idMap, this.obj.getMember("_FileCountLimitExceededException__limit")));
    return idMap;
  }
}
